types/sparqljs/index.d.ts

```diff
@@ -1,303 +1,311 @@
- *   SELECT (?a as ?b) ... ==> { expression: '?a', variable: '?b' }
- */
-export type Variable = VariableExpression | VariableTerm;
-
-export interface VariableExpression {
-    expression: Expression;
-    variable: VariableTerm;
-}
-
-export type Pattern =
-    | BgpPattern
-    | BlockPattern
-    | FilterPattern
-    | BindPattern
-    | ValuesPattern
-    | SelectQuery;
-
-/**
- * Basic Graph Pattern
- */
-export interface BgpPattern {
-    type: "bgp";
-    triples: Triple[];
-}
-
-export interface GraphQuads {
-    type: "graph";
-    name: IriTerm;
-    triples: Triple[];
-}
-
-export type BlockPattern =
-    | OptionalPattern
-    | UnionPattern
-    | GroupPattern
-    | GraphPattern
-    | MinusPattern
-    | ServicePattern;
-
-export interface OptionalPattern {
-    type: "optional";
-    patterns: Pattern[];
-}
-
-export interface UnionPattern {
-    type: "union";
-    patterns: Pattern[];
-}
-
-export interface GroupPattern {
-    type: "group";
-    patterns: Pattern[];
-}
-
-export interface GraphPattern {
-    type: "graph";
-    name: IriTerm;
-    patterns: Pattern[];
-}
-
-export interface MinusPattern {
-    type: "minus";
-    patterns: Pattern[];
-}
-
-export interface ServicePattern {
-    type: "service";
-    name: IriTerm;
-    silent: boolean;
-    patterns: Pattern[];
-}
-
-export interface FilterPattern {
-    type: "filter";
-    expression: Expression;
-}
-
-export interface BindPattern {
-    type: "bind";
-    expression: Expression;
-    variable: VariableTerm;
-}
-
-export interface ValuesPattern {
-    type: "values";
-    values: ValuePatternRow[];
-}
-
-export interface ValuePatternRow {
-    [variable: string]: IriTerm | BlankTerm | LiteralTerm | undefined;
-}
-
-export interface Triple {
-    subject: IriTerm | BlankTerm | VariableTerm | QuadTerm;
-    predicate: IriTerm | VariableTerm | PropertyPath;
-    object: Term;
-}
-
-export interface PropertyPath {
-    type: "path";
-    pathType: "|" | "/" | "^" | "+" | "*" | "!";
-    items: Array<IriTerm | PropertyPath>;
-}
-
-export type Expression =
-    | OperationExpression
-    | FunctionCallExpression
-    | AggregateExpression
-    | BgpPattern
-    | GraphPattern
-    | GroupPattern
-    | Tuple
-    | Term;
-
-// allow Expression circularly reference itself
-export interface Tuple extends Array<Expression> {}
-
-export interface BaseExpression {
-    type: string;
-    distinct?: boolean | undefined;
-}
-
-export interface OperationExpression extends BaseExpression {
-    type: "operation";
-    operator: string;
-    args: Expression[];
-}
-
-export interface FunctionCallExpression extends BaseExpression {
-    type: "functionCall";
-    function: string;
-    args: Expression[];
-}
-
-export interface AggregateExpression extends BaseExpression {
-    type: "aggregate";
-    expression: Expression;
-    aggregation: string;
-    separator?: string | undefined;
-}
+// tslint:disable-next-line:no-unnecessary-class
+declare class SparqlJs {
```

Comment:
This doesn't look right; upstream has `module.exports = { ... }` with just three things, not a class exported with extra props: https://unpkg.com/browse/sparqljs@3.7.1/sparql.js

If this is also related to `import/default`, I'm pretty sure that lint rule is bugged if it's complatining. https://arethetypeswrong.github.io/?p=sparqljs%403.7.1 is happy.

Reply:
I meant to comment on `export =`, oops. Still, point stands.